src/account.rs

```diff
@@ -302,15 +310,25 @@ impl Account {
         ]))
             .try_into()?;
 
-        let mut mileage = mileage
-            .min(self.daily - self.day)
-            .min(self.weekly - self.week)
-            .min(self.end);
+        let maximum = if Local::now().naive_local().ordinal() == datetime.ordinal() {
+            (self.daily - self.day)
+                .min(self.weekly - self.week)
+                .min(self.end)
+        } else {
+            self.end.min(self.daily).min(self.weekly)
+        };
```

Comment:
I try to make a simple judgement for maximum mileage temporarily if the given time is today to avoid it's unable to send a submission of running data if today had already runned enough time.

```rust
let maximum = if Local::now().naive_local().ordinal() == datetime.ordinal() {
    (self.daily - self.day)
        .min(self.weekly - self.week)
        .min(self.end)
} else {
    self.end.min(self.daily).min(self.weekly)
};
let mut mileage = maximum * percent / 100.;
```

No doubt this is a temporary solution that may need to be modified later when your new interface captured is interfaced to this project.

Reply:
Don't think this will do the trick. Since the default time is set when the `Tui` is created and it won't match the time when we call the method.

Reply:
Maybe make the `datetime` an `Option` and get the current time when it is `None`?
Idk, I wasn't gonna continue completing this project since the most important dependency `tui-rs` is already archived and replaced by `rata-tui`. If we are going to add more features maybe it is better to just refactor it.

Reply:
> Don't think this will do the trick. Since the default time is set when the `Tui` is created and it won't match the time when we call the method.

Obviously this is a callback function, and the current time is determined by `Local::now()`, which means it really doesn't affect when the program is started.

Reply:
> Maybe make the `datetime` an `Option` and get the current time when it is `None`? Idk, I wasn't gonna continue completing this project since the most important dependency `tui-rs` is already archived and replaced by `rata-tui`. If we are going to add more features maybe it is better to just refactor it.

Emm, Do you think it's a good idea to build directly with `Tauri`? Although Tauri 2 is still in beta, I haven't had any problems so far. And it's esay to develop.

Reply:
> Emm, Do you think it's a good idea to refactor this project directly with Tauri? Although Tauri 2 is still in beta, I haven't had any problems so far. And it's esay to develop.

Well, I do not think it will ever need Tauri since such a simple logic can easily be achieved in the web. All in all it's all about frontend things and I am going for the Leptos way. Feel free to Wrap it in Tauri if you prefer a GUI client but I am not going to pay another space on my device for such an app.

Reply:
> All in all it's all about frontend things and I am going for the Leptos way

Actually, if we refactor with Leptos and successfully compile it to WASM, we could even deploy it on GitHub Pages. However a client side app would still be necessary for anyone requiring periodic execution, but I would keep it simple and make a CLI one.